plugins/poll/db/migrate/20180820080623_migrate_polls_data.rb

```diff
@@ -0,0 +1,82 @@
+class MigratePollsData < ActiveRecord::Migration[5.2]
+  def up
+    execute <<~SQL
```

Comment:
I am somewhat worried about this, in particular how does this handle **corrupt** data, we need to make sure that there is no case where we run this migration and just explode. I would kind of prefer if this iterated through all polls and perform the conversion in ruby just cause of the safety aspect.

Reply:
We'll also need to think about how we're going to handle the deployment process. After this migration has run, the old custom fields are still writable which means that a poll created or vote made before the app servers have been deployed with the new code will be lost. Iterating through the polls and doing the migration in Ruby might be safer too as we can acquire a lock around the poll while the migration is happening.

Reply:
@SamSaffron I was worried about that too and tried my best to handle this in the query. But I guess doing it in Ruby is safer. Will change.

@tgxworld I personally am fine losing votes during the migration and the odds of people making a poll while the migration is running are very very low. I'd much rather lose a bit of data than add a lot more complexity to this migration.

Reply:
> during the migration and the odds of people making a poll while the migration is running are very very low

It isn't the duration of the migration that results in the window of the data loss. The window of data loss is open when the migration has run but the app servers have not been redeployed with the latest code. That window can be quite significant depending on the different build processes we have in place.

Reply:
New migration is in. It's still a monster. Effectively slower but should be more safe.